compute/oslogin/service_account_ssh.py

```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example of using the OS Login API to apply public SSH keys for a service
+account, and use that service account to execute commands on a remote
+instance over SSH. This example uses zonal DNS names to address instances
+on the same internal VPC network.
+"""
+
+# [START imports_and_variables]
+import time
+import subprocess
+import uuid
+import logging
+import requests
+import argparse
+import googleapiclient.discovery
+
+# Global variables
+SERVICE_ACCOUNT_METADATA_URL = (
+    'http://metadata.google.internal/computeMetadata/v1/instance/'
+    'service-accounts/default/email')
+HEADERS = {'Metadata-Flavor': 'Google'}
+
+# [END imports_and_variables]
+
+
+# [START run_command_local]
+def execute(cmd, cwd=None, capture_output=False, env=None, raise_errors=True):
+    """Execute an external command (wrapper for Python subprocess)."""
+    logging.info('Executing command: {cmd}'.format(cmd=str(cmd)))
+    stdout = subprocess.PIPE if capture_output else None
+    process = subprocess.Popen(cmd, cwd=cwd, env=env, stdout=stdout)
+    output = process.communicate()[0]
+    returncode = process.returncode
+    if returncode:
+        # Error
+        if raise_errors:
+            raise subprocess.CalledProcessError(returncode, cmd)
+        else:
+            logging.info('Command returned error status %s', returncode)
+    if output:
+        logging.info(output)
+    return returncode, output
+# [END run_command_local]
+
+
+# [START create_key]
+def create_ssh_key(oslogin, account, private_key_file=None, expire_time=300):
+    """Generate an SSH key pair and apply it to the specified account."""
+    private_key_file = private_key_file or '/tmp/key-' + uuid.uuid4()
```

Comment:
uuid4() returns a UUID object that can't be concatenated with a string. Try str(uuid.uuid4()) to fix that.

Reply:
Done.